python/runNanoSeq.py

```diff
@@ -0,0 +1,938 @@
+        arow = row.strip().split('\t')
+        if csvFiles.get(arow[0], None):
+          csvFiles[arow[0]].write('%s\n' % ','.join(arow[1:]))
+    
+    for ifile in csvFiles.values():
+      ifile.close()
+
+    print("\nMerge coverage files for var\n")
+    #merge coverage files
+    outFile = "%s/post/cov.bed"%(tmpDir )
+    cmd = "rm -f %s;"%( outFile)
+    for i in range(nfiles ) :
+      ifile = "%s/var/%s.cov.bed.gz"%(tmpDir, i )
+      cmd += "bgzip -dc %s >> %s ;"%( ifile, outFile )
+    cmd += "bgzip -@ %s -f %s; sleep 3; bgzip -@ %s -t %s.gz ;"%(args.threads,outFile,args.threads,outFile)
+    cmd += "tabix -f %s.gz"%(outFile)
+    runCommand( cmd )
+
+    print("\nCompute summaries for var\n")
+    #do the summary
+    cmd = "variantcaller.R %s/post/ > %s/post/summary.txt"%(tmpDir,tmpDir)
+    runCommand( cmd )
+
+    cmd = "nanoseq_results_plotter.R %s/post %s/post/results"%(tmpDir,tmpDir)
+    runCommand( cmd )
+
+    print("\nGenerate vcf file from variants.csv\n")
+
+    var = {}
+    nVariants = 0
+    with open("%s/post/variants.csv"%tmpDir,'r') as iofile :
+      iline = iofile.readline().rstrip('\n')
+      fields = iline.split(',')
+      for ifield in fields :
+        var[ ifield ] = []
+      for iline in iofile :
+        nVariants += 1
+        for (i, ival) in enumerate(iline.rstrip('\n').split(',')) :
+          var[fields[i]].append(ival)
+      
+    header =  '##fileformat=VCFv4.2\n'
+    header += '##source=NanoSeq pipeline\n'
+    header += '##FILTER=<ID=PASS,Description="All filters passed">\n'
+    header += "##reference=file://%s\n"%args.ref
+    with open(args.ref + '.fai','r') as iofile :
+      for iline in iofile :
+        ichr = iline.split('\t')[0]
+        ilength = iline.split('\t')[1]
+        header += "##contig=<ID=%s,length=%s>\n"%(ichr,ilength)
+    header += '##ALT=<ID=*,Description="Represents allele(s) other than observed.">\n'
+    header += '##INFO=<ID=TRI,Number=1,Type=String,Description="Pyrimidine context, trinucleotide substitution">\n'
+    header += '##INFO=<ID=BBEG,Number=1,Type=String,Description="Read bundle left breakpoint">\n'
+    header += '##INFO=<ID=BEND,Number=1,Type=String,Description="Read bundle right breakpoint">\n'
+    header += '##INFO=<ID=QPOS,Number=1,Type=Integer,Description="Read position closest to 5-prime end">\n'
+    header += '##INFO=<ID=DEPTH_FWD,Number=1,Type=Integer,Description="Read bundle forward reads depth">\n'
+    header += '##INFO=<ID=DEPTH_REV,Number=1,Type=Integer,Description="Read bundle reverse reads depth">\n'
+    header += '##INFO=<ID=DEPTH_NORM_FWD,Number=1,Type=Integer,Description="Matched normal forward reads depth">\n'
+    header += '##INFO=<ID=DEPTH_NORM_REV,Number=1,Type=Integer,Description="Matched normal reverse reads depth">\n'
+    header += '##FILTER=<ID=dbsnp,Description="Common SNP site">\n'
+    header += '##FILTER=<ID=shearwater,Description="Noisy site">\n'
+    header += '#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n'
```

Comment:
Move all the header generation into a function that returns a list of strings and then use `"\n".join(header_lines)` to write to the file.  This will aid readability of this very large function.